builders/interoperability/xcm/xc-registration/foreign-asset-registration.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Self-Serve Asset Registration for Sibling Parachains
+description: This guide shows sibling parachains how to register native tokens as foreign assets on Moonbeam via ForeignAssetOwnerOrigin to unlock ERC-20 UX on Moonbeam.
+---
+
+# Self-Serve Asset Registration for Sibling Parachains
+
+## Introduction {: #introduction }
+
+Registering your parachain’s native tokens on Moonbeam or Moonriver lets your community enjoy ERC‑20–style UX and deep EVM integrations while retaining full on‑chain provenance. This guide shows sibling Polkadot parachain teams how to self‑register a foreign asset using the new `ForeignAssetOwnerOrigin` introduced in Moonbeam Runtime 3600.
+
+## Why a New Origin? {: #why-a-new-origin }
+
+Moonbeam introduced a new dedicated origin called `ForeignAssetOwnerOrigin`, which only permits an XCM message whose origin contains the asset’s MultiLocation to execute calls in the `evm‑foreign‑assets` pallet. In practice, that means only the sovereign account of the parachain that owns the asset, or Moonbeam governance, can create, freeze, unfreeze, or relocate it. Alongside this, a configurable runtime constant called `ForeignAssetCreationDeposit` is reserved from the caller’s sovereign account at creation time. The deposit discourages spam registrations.
+
+## Required Deposits {: #required-deposits }
+
+To prevent spam, a `ForeignAssetCreationDeposit` is required and locked for the asset's lifetime. The deposit is funded from the sibling parachain's sovereign account on the Moonbeam network, which thus needs to be sufficiently funded to cover the asset deposit and the associated transaction fees. If the asset is destroyed through governance, the deposit is unreserved and returned to the original sovereign account.
+
+Deposits are network‑specific and can be adjusted by Moonbeam governance via the `parameters` pallet:
+
+=== "Moonbeam"
+
+     ```text
+     {{ networks.moonbeam.xcm.foreign_asset_deposit.display }} GLMR
+     ```
+
+=== "Moonriver"
+
+     ```text
+     {{ networks.moonriver.xcm.foreign_asset_deposit.display }} MOVR
+     ```
+
+=== "Moonbase Alpha"
+
+     ```text
+     {{ networks.moonbase.xcm.foreign_asset_deposit.display }} DEV
+     ```
+
+## Prerequisites {: #prerequisites }
```

Comment:
Should we consider a prerequisite to have a bidirectional XCM channel with Moonbeam established (we need one!)